### ISSUE TITLE
ScrollBar resize can cause unintentional scroll

### ISSUE THREAD

**Issue by moniika** [google/blockly#4192]
**Describe the bug**

The scrollbar `resize` call can unintentionally scroll.

**To Reproduce**

1. Add 4 blocks with space between them and prepare a 5th block that would fit in empty space.
```
Blockly.Xml.domToWorkspace(
    Blockly.Xml.textToDom(
       '<xml xmlns="https://developers.google.com/blockly/xml">' +
       '<block type="controls_if" x="88" y="88"></block>' +
        '<block type="controls_if" x="288" y="88"></block>' +
        '<block type="controls_if" x="88" y="238"></block>' +
        '<block type="controls_if" x="288" y="238"></block>' +
        '</xml>'),
    workspace);
var xmlDom = Blockly.Xml.textToDom(
    '<xml xmlns="https://developers.google.com/blockly/xml">' +
    '<block type="controls_if" x="188" y="163"></block> +
    '</xml>');
```
2. Set breakpoints at `onScroll()` call in `Scrollbar.prototyle.resize` and in `WorkspaceSvg.setTopLevelWorkspaceMetrics_`
3. Add block in the middle of the existing blocks, should not trigger any scrolling.
```
Blockly.Xml.domToWorkspace(xmlDom, workspace);
```
4. Step through breakpoints observing that `hostmetrics.viewLeft` is 0 in `Scrollbar resize`, but in `setTopLevelWorkspaceMetrics_` `this.scrollY` is set to small but non-zero value, where we would expect zero.

**Expected behavior**

There shouldn't be scrolling.

**Additional context**

This can cause unwanted extra emission of viewport change events, once they are enabled.

**Comment by rachel-fenichel** [comment]
Low priority because most of the possible triggers are blocked by `workspace.setResizeEnabled(false)`, such as during drags.
